tests/python/test_usdt.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/python
+#
+# USAGE: test_usdt.py
+#
+# Copyright 2017 Facebook, Inc
+# Licensed under the Apache License, Version 2.0 (the "License")
+
+from __future__ import print_function
+from bcc import BPF, USDT
+import sys
+import os
+import signal
+import subprocess
+import tempfile
+
+# Application, minimum, to define three trace points
+app_text = """
+#include <unistd.h>
+#include <sys/sdt.h>
+
+int main() {
+  char s[100];
+  int i, a = 20, b = 40;
+  for (i = 0; i < 100; i++) s[i] = (i & 7) + (i & 6);
+  while (1) {
+    DTRACE_PROBE2(test, probe_point_1, s[7], b);
+    DTRACE_PROBE2(test, probe_point_3, a, b);
+    sleep(3);
+    a++; b++;
+    DTRACE_PROBE2(test, probe_point_1, s[4], a);
+    DTRACE_PROBE2(test, probe_point_2, 5, s[10]);
+    DTRACE_PROBE2(test, probe_point_3, s[4], s[7]);
+  }
+  return 1;
+}
+"""
+
+# Compile and run the application
+ftemp = tempfile.NamedTemporaryFile(delete=False, suffix=".c")
+ftemp.write(app_text)
+ftemp.close()
+subprocess.check_call(["gcc", "-I", "%s/include" % os.getcwd(), "-o", "%s.out" % ftemp.name, ftemp.name])
```

Comment:
Can you use a pipe instead of a named temp file? I expect you'll still want it for the output binary, but 1 is better than 2.

See https://stackoverflow.com/questions/16552794/piping-a-string-to-gcc-with-pythons-subprocess-popen for example.